vmdb/app/models/vm_microsoft.rb

```diff
@@ -7,4 +7,8 @@ def self.calculate_power_state(raw_power_state)
     else                        super
     end
   end
+
+  def supports_migrate?
+    false
+  end
```

Comment:
That's exactly why it's confusing.  It's a terrible design to have the caller filter things out, and then assume that the following model methods "work" just because they weren't called.  *Someone* is going to try calling the method on templates, and the results will be wrong.  That link you showed could be completely eliminated if all of the objects respond to their capabilities properly (including clone and migrate and others)